### ISSUE TITLE
podman machine start reports an error although it starts, volumes not mounted to machine

### ISSUE THREAD

**Issue by esthomw** [containers/podman#20081]
### Issue Description

There a are two issues that might be related:
* `podman machine start` logs `Error: machine did not transition into running state` although I can ssh to the machine and I can run containers just fine,
* after createing the machine with `-v $HOME:$HOME` the volume is not actually mounted, checked on machine via ssh. This causes `Error: statfs ... no such file or directory` issues.

### Steps to reproduce the issue

❯ podman version
Client:       Podman Engine
Version:      4.6.2
API Version:  4.6.2
Go Version:   go1.21.0
Git Commit:   5db42e86862ef42c59304c38aa583732fd80f178
Built:        Mon Aug 28 16:55:03 2023
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.6.2
API Version:  4.6.2
Go Version:   go1.20.7
Built:        Tue Sep 12 22:07:26 2023
OS/Arch:      linux/arm64

### Describe the results you received

False positive error when starting a machine.
Volumes not mounted to the machine.

### Describe the results you expected

I expect machine to start without errors with volumes mounted as specified via cli `machine init` flags.

### podman info output

```yaml
❯ podman info
host:
  arch: arm64
  buildahVersion: 1.31.2
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.7-2.fc38.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 99.57
    systemPercent: 0.3
    userPercent: 0.12
  cpus: 4
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: coreos
    version: "38"
  eventLogger: journald
  freeLocks: 2048
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
    uidmap:
    - container_id: 0
      host_id: 502
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
  kernel: 6.4.15-200.fc38.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 7731699712
  memTotal: 8199520256
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.7.0-1.fc38.aarch64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.7.0
    package: netavark-1.7.0-1.fc38.aarch64
    path: /usr/libexec/podman/netavark
    version: netavark 1.7.0
  ociRuntime:
    name: crun
    package: crun-1.9-1.fc38.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.9
      commit: a538ac4ea1ff319bcfe2bf81cb5c6f687e2dc9d3
      rundir: /run/user/502/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20230908.g05627dc-1.fc38.aarch64
    version: |
      pasta 0^20230908.g05627dc-1.fc38.aarch64-pasta
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/502/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.1-1.fc38.aarch64
    version: |-
      slirp4netns version 1.2.1
      commit: 09e31e92fa3d2a1d3ca261adaeb012c8d75a8194
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 0h 8m 37.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphRootAllocated: 106769133568
  graphRootUsed: 4055769088
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/502/containers
  transientStore: false
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 4.6.2
  Built: 1694549246
  BuiltTime: Tue Sep 12 22:07:26 2023
  GitCommit: ""
  GoVersion: go1.20.7
  Os: linux
  OsArch: linux/arm64
  Version: 4.6.2
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by shankarpnsn** [comment]
I'm having the same issue with podman version 4.6.2. The workaround suggested in [https://github.com/containers/podman/issues/17403#issuecomment-1536636874] isn't working anymore.

**Comment by baude** [comment]
any chance this is on a rasp pi or something that has really slow io?  more information would be helpful.  also maybe include the start with `--log-level=debug`.  And maybe check for any running gvproxy processes prior ?

**Comment by nastacio** [comment]
I started hitting the same problem on macOS Ventura 13.4.1 with podman 4.6.2:

```
podman version
Client:       Podman Engine
Version:      4.6.2
API Version:  4.6.2
Go Version:   go1.21.0
Git Commit:   5db42e86862ef42c59304c38aa583732fd80f178
Built:        Mon Aug 28 10:55:03 2023
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.6.2
API Version:  4.6.2
Go Version:   go1.21.0
Built:        Tue Sep 12 15:59:42 2023
OS/Arch:      linux/arm64
```

VM started with:

```
 podman machine init --cpus=2 --memory=4096 --image-path=testing --now  --log-level=debug
INFO[0000] podman filtering at log level debug          
DEBU[0000] Using Podman machine with `qemu` virtualization provider 
Extracting compressed file
Image resized.
Machine init complete
DEBU[0019] Using Podman machine with `qemu` virtualization provider 
Starting machine "podman-machine-default"
[/opt/homebrew/opt/podman/libexec/podman/gvproxy -listen-qemu unix:///var/folders/qh/3_60ws154z9_3yvgxb29n7z80000gn/T/podman/qmp_podman-machine-default.sock -pid-file /var/folders/qh/3_60ws154z9_3yvgxb29n7z80000gn/T/podman/podman-machine-default_proxy.pid -ssh-port 53825 -forward-sock /Users/nastacio/.local/share/containers/podman/machine/qemu/podman.sock -forward-dest /run/user/501/podman/podman.sock -forward-user core -forward-identity /Users/nastacio/.ssh/podman-machine-default --debug]
DEBU[0019] qemu cmd: [/opt/homebrew/bin/qemu-system-aarch64 -m 4096 -smp 2 -fw_cfg name=opt/com.coreos/config,file=/Users/nastacio/.config/containers/podman/machine/qemu/podman-machine-default.ign -qmp unix:/var/folders/qh/3_60ws154z9_3yvgxb29n7z80000gn/T/podman/qmp_podman-machine-default.sock,server=on,wait=off -netdev socket,id=vlan,fd=3 -device virtio-net-pci,netdev=vlan,mac=5a:94:ef:e4:0c:ee -device virtio-serial -chardev socket,path=/var/folders/qh/3_60ws154z9_3yvgxb29n7z80000gn/T/podman/podman-machine-default_ready.sock,server=on,wait=off,id=apodman-machine-default_ready -device virtserialport,chardev=apodman-machine-default_ready,name=org.fedoraproject.port.0 -pidfile /var/folders/qh/3_60ws154z9_3yvgxb29n7z80000gn/T/podman/podman-machine-default_vm.pid -accel hvf -accel tcg -cpu host -M virt,highmem=on -drive file=/opt/homebrew/share/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on -drive file=/Users/nastacio/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd,if=pflash,format=raw -virtfs local,path=/Users,mount_tag=vol0,security_model=none -virtfs local,path=/private,mount_tag=vol1,security_model=none -virtfs local,path=/var/folders,mount_tag=vol2,security_model=none -drive if=virtio,file=/Users/nastacio/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-38.20230918.2.0-qemu.aarch64.qcow2] 
Waiting for VM ...
```
Notice the QEMU window output waiting on a login:

<img width="641" alt="image" src="https://github.com/containers/podman/assets/3278623/609656f2-a0bc-4722-aaa8-80e882db87a1">

```
podman machine inspect
[
     {
          "ConfigPath": {
               "Path": "/Users/nastacio/.config/containers/podman/machine/qemu/podman-machine-default.json"
          },
          "ConnectionInfo": {
               "PodmanSocket": {
                    "Path": "/Users/nastacio/.local/share/containers/podman/machine/qemu/podman.sock"
               },
               "PodmanPipe": null
          },
          "Created": "2023-09-28T09:42:32.018705-04:00",
          "Image": {
               "IgnitionFilePath": {
                    "Path": "/Users/nastacio/.config/containers/podman/machine/qemu/podman-machine-default.ign"
               },
               "ImageStream": "testing",
               "ImagePath": {
                    "Path": "/Users/nastacio/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-38.20230918.2.0-qemu.aarch64.qcow2"
               }
          },
          "LastUp": "0001-01-01T00:00:00Z",
          "Name": "podman-machine-default",
          "Resources": {
               "CPUs": 2,
               "DiskSize": 100,
               "Memory": 4096
          },
          "SSHConfig": {
               "IdentityPath": "/Users/nastacio/.ssh/podman-machine-default",
               "Port": 53825,
               "RemoteUsername": "core"
          },
          "State": "running",
          "UserModeNetworking": true
     }
]
```

**Comment by nastacio** [comment]
I get the same results regardless of image-path: `stable`, `testing`, `next`.

**Comment by esthomw** [comment]
I tried setting things up from scratch, here is what happened:
```
podman machine init --cpus 4 --memory 8096 -v "$HOME/projects:$HOME/projects"
```
This command managed to create the machine.
```
podman machine start --log-level=debug
```
During first start everything went smooth. The machine successfully started. I tried running an image:
```
podman run -p 8080:8080 -v $HOME/projects/airflow-dev/dags:/opt/airflow/dags docker.io/apache/airflow standalone
```
and this succeeded as well. Afterwards I stopped the image, stopped the machine and tried repeating the whole process. This time, when I tried starting the machine, I got the following output (`--log-level=debug` enabled, I replaced path part with `$HOME` in the output):
```
INFO[0000] podman filtering at log level debug          
DEBU[0000] Using Podman machine with `qemu` virtualization provider 
Starting machine "podman-machine-default"
[/opt/homebrew/opt/podman/libexec/podman/gvproxy -listen-qemu unix:///var/folders/6k/sdcpjks958j1h489_9cbll6w0000gp/T/podman/qmp_podman-machine-default.sock -pid-file /var/folders/6k/sdcpjks958j1h489_9cbll6w0000gp/T/podman/podman-machine-default_proxy.pid -ssh-port 50459 -forward-sock $HOME/.local/share/containers/podman/machine/qemu/podman.sock -forward-dest /run/podman/podman.sock -forward-user root -forward-identity $HOME/.ssh/podman-machine-default --debug]
DEBU[0000] qemu cmd: [/opt/homebrew/bin/qemu-system-aarch64 -m 8096 -smp 4 -fw_cfg name=opt/com.coreos/config,file=$HOME/.config/containers/podman/machine/qemu/podman-machine-default.ign -qmp unix:/var/folders/6k/sdcpjks958j1h489_9cbll6w0000gp/T/podman/qmp_podman-machine-default.sock,server=on,wait=off -netdev socket,id=vlan,fd=3 -device virtio-net-pci,netdev=vlan,mac=5a:94:ef:e4:0c:ee -device virtio-serial -chardev socket,path=/var/folders/6k/sdcpjks958j1h489_9cbll6w0000gp/T/podman/podman-machine-default_ready.sock,server=on,wait=off,id=apodman-machine-default_ready -device virtserialport,chardev=apodman-machine-default_ready,name=org.fedoraproject.port.0 -pidfile /var/folders/6k/sdcpjks958j1h489_9cbll6w0000gp/T/podman/podman-machine-default_vm.pid -accel hvf -accel tcg -cpu host -M virt,highmem=on -drive file=/opt/homebrew/share/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on -drive file=$HOME/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd,if=pflash,format=raw -virtfs local,path=$HOME/projects,mount_tag=vol0,security_model=none -drive if=virtio,file=$HOME/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-38.20230918.2.0-qemu.aarch64.qcow2] 
Waiting for VM ...
Error: machine did not transition into running state
DEBU[0029] Shutting down engines 
```

QEMU log show exactly the same message that @nastacio pasted in this thread:
<img width="637" alt="Screenshot 2023-09-29 at 09 30 37" src="https://github.com/containers/podman/assets/39091820/fa0e105d-2647-40b2-a6fb-d0413562bb87">

**Comment by baude** [comment]
is anyone seeing this with podman 4.7?

**Comment by esthomw** [comment]
@baude currently I am on:
```
Client:       Podman Engine
Version:      4.7.1
API Version:  4.7.1
Go Version:   go1.21.3
Git Commit:   ef83eeb9c7482826672f3efa12db3d61c88df6c4
Built:        Thu Oct  5 15:21:06 2023
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.7.0
API Version:  4.7.0
Go Version:   go1.20.8
Built:        Wed Sep 27 20:24:25 2023
OS/Arch:      linux/arm64
```

and the issue is still present.

**Comment by ssbarnea** [comment]
@baude Yes, I do with 4.7.1 and with previous versions I did not even had to add any mounting volumes to the init command as they were supposed to be added automatically on macos (sonoma/14.0). Apparently they are no longer added. Only note is that I see it downloading fedora-core-39 image.

I am not sure when the regression happened but it was recently (last two months). The previous image which I removed was working was about two months old.
